AutomatedTesting/Gem/PythonTests/Physics/tests/EntityComponentTests/PhysX_Collider_Component_CRUD.py

```diff
@@ -0,0 +1,167 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+def PhysX_Collider_Component_CRUD():
+    """
+    Summary:
+    Creating a PhysX Collider and setting values to all properties that are supported via the Editor Context Bus.
+
+    Expected Behavior:
+    Validates that Component CRUD can be performed without crashing.
+
+    PhysX Collider Components:
+     - PhysX Collider
+
+    Test Steps:
+     1) Add an Entity to manipulate
+     2) Add a PhysX Collider Component to Entity
+     3) Set Box Shape and Child Properties
+     4) Set Capsule Shape and Child Properties
+     5) Set Cylinder Shape and Child Properties
+     6) Set Sphere Shape and Child Properties
+     7) Set General Properties
+     8) Set PhyicsAsset Shape and Child Properties
+     9) Delete Component
+
+    :return: None
+    """
+
+    # Helper file Imports
+    import os
+
+    from editor_python_test_tools.editor_component.physx_collider import PhysxCollider
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Tracer, TestHelper
+
+    from consts.general import Strings
+
+    # 0) Pre-conditions
+    physx_mesh = os.path.join("objects", "_primitives", "_box_1x1.pxmesh")
+    physx_material = os.path.join("physx", "glass.physxmaterial")
+
+    TestHelper.init_idle()
+    TestHelper.open_level("", "Base")
+
+    with Tracer() as section_tracer:
+    # 1 ) Add an Entity to manipulate
+        phsyx_collider_entity = EditorEntity.create_editor_entity("TestEntity")
+
+    # 2) Add a PhysX Collider Component to Entity
+        physx_collider = PhysxCollider(phsyx_collider_entity)
+
+    # 3) Set Box Shape and Child Properties
+        physx_collider.set_box_shape()
+
+        physx_collider.set_box_dimensions(256.0, 256.0, 256.0)
+        physx_collider.set_box_dimensions(2.5, 2.5, 2.5)
+        physx_collider.set_box_dimensions(0.0, 0.0, 0.0)
+        physx_collider.set_box_dimensions(-1.0, -10, -256.0)
+
+    # 4) Set Capsule Shape and Child Properties
+        physx_collider.set_capsule_shape()
+
+        physx_collider.set_capsule_height(0)
+        physx_collider.set_capsule_height(-1.0)
+        physx_collider.set_capsule_height(2.5)
+        physx_collider.set_capsule_height(256.0)
```

Comment:
none of this test code makes any actual test assertions it's all shoved into the utilities. the test script doesn't say to me I expect the value to be set correctly it's reading like something else is missing to me.

Reply:
Refactored